src/desktop/dynamic_launcher.rs

```diff
@@ -321,9 +321,15 @@ impl<'a> DynamicLauncherProxy<'a> {
     /// See also [`Launch`](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.DynamicLauncher.html#org-freedesktop-portal-dynamiclauncher-launch).
     #[doc(alias = "Launch")]
     #[doc(alias = "xdp_portal_dynamic_launcher_launch")]
-    pub async fn launch(&self, desktop_file_id: &str) -> Result<(), Error> {
-        // TODO: handle activation_token
-        let options: HashMap<&str, zvariant::Value<'_>> = HashMap::new();
+    pub async fn launch(
+        &self,
+        desktop_file_id: &str,
+        activation_token: impl Into<Option<String>>,
+    ) -> Result<(), Error> {
+        let mut options: HashMap<&str, zvariant::Value<'_>> = HashMap::new();
+        if let Some(activation_token) = activation_token.into() {
+            options.insert("activation_token", activation_token.into());
+        }
```

Comment:
This should be a struct similar to other options we have in the library.

Reply:
But also I am not sure we want to expose the activation token as a random string, I would instead make it a new type that we could potentially validate

Reply:
I know that you can't really validate an activation token under wayland but I don't know if it is the case for x11. Also, we would ideally expose in the future a way to create an activation token from a wayland / x11 surface, similar to how we do for the WindowIdentifier. See #70

Reply:
Looking at the validation part, it seems like neither Wayland nor [X11](https://github.com/rust-windowing/winit/blob/db2c97a995bc403e0442675c7364010b806d5431/src/platform_impl/linux/x11/activation.rs#L19) have a unified standard for the token. Should I still add an `ActivationToken` type anyway?

Reply:
I would still define a new type as we would need to provide constructors and so on for later. For the validation part, it would be nice to document that both x11 and wayland activation tokens can't be validated, so just implement From<String> and other traits.